### PR TITLE
Change CSS files path to RTL stylesheets if needed

### PR DESCRIPTION
**Pull request by ramtin2025**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | "rtl"
| Description?  | Change CSS files path to RTL stylesheets if needed
| Type?         | new feature
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Select RTL language on BO or FO.

This PR contains:
- Modified StylesheetManager.php to support RTL

### CODE REVIEW

**Comment by mehrshadz** [comment]
@maximebiloe 
@xBorderie 
@AlexEven 
We don't have any problem with "StylesheetManager.php" file. What's the "Travis" error for?

**Comment by maximebiloe** [comment]
Hello @ramtin2025 & @mehrshadz,

As you change the function signature, the tests in `AssetsCoreTest` class are broken. Indeed, you're setting the `$needRtl` to `true` by default, the assets are not generated. So, you need to rewrite the tests to fix them.

Regards

**Comment by AlexEven** [comment]
Hello @ramtin2025 & @mehrshadz!

We need you to edit the test as Maxime mentioned, could you please have a look?

You also should squash your commits so that we only have "CO: Change CSS files path to RTL stylesheets if needed" as a commit name (the other commits aren't respecting our standards). You should avoid editing directly code in GitHub as a general rule.
 
That way your PR will be valid and we can merge it.

Let us know if we can help!
Thanks!

File: classes/assets/StylesheetManager.php [hunk 1]
```diff
@@ -53,11 +53,18 @@ public function register(
...
+        $needRtl = true
     ) {
+        $fullPath = $this->getFullPath($relativePath);
+        $rtlFullPath = $this->getFullPath(str_replace('.css', '_rtl.css', $relativePath));
+        $context = Context::getContext();
+        $isRTLed = $context->language->is_rtl;
         if ('remote' === $server) {
             $this->add($id, $relativePath, $media, $priority, $inline, $server);
-        } elseif ($fullPath = $this->getFullPath($relativePath)) {
+        } else if ($needRtl && $isRTLed && $rtlFullPath) {
```
Comment (Danoosh) [inline classes/assets/StylesheetManager.php]: @maximebiloe I've checked AssetsCoreTest and I can't find any problem in it. needRTL is true by default but another condition is needed to make it works.

**Comment by Danoosh** [comment]
@maximebiloe I've checked AssetsCoreTest and I can't find any problem in it. needRTL is true by default but another condition is needed to make it work.

else if ($needRtl && $isRTLed && $rtlFullPath)

Update: AssetsCoreTest must have context language, however, I've changed my code.

**Comment by mehrshadz** [comment]
@maximebiloe 
We've fixed related Travis error

regards

**Review by maximebiloe — APPROVED** [review]

**Comment by maximebiloe** [comment]
Thank you @ramtin2025

**Pull request merged**